corn-maze.lic

```diff
@@ -0,0 +1,632 @@
+    wait_for_command
+  end
+
+  def find_start
+    wait_on_done_move
+    unless XMLData.room_title.include?('Labyrinth') || XMLData.room_title.include?('Corn Maze, Entrance')
+      DRC.message("The script can't find its way back from here!  Make your way back towards the northwest Labyrinth using elanthipedia's Corn Maze entry, such as https://elanthipedia.play.net/Sleeping_Dragon_Corn_Maze_432 and then unpause the script!")
+      pause_script('cornmaze')
+    end
+    until @done
+      exit if Flags['maze_done']
+      find_exits = DRC.bput("look",/Obvious paths: .*\./,)
+      exits = find_exits.match(/^Obvious paths: (.*)\./).to_a[1].split(', ')
+      if (exits.include? 'north')
+        custom_move 'north'
+      elsif (exits.include? 'east')
+        custom_move 'east'
+      elsif (exits.include? 'south')
+        custom_move 'south'
+        custom_move 'south'
+        custom_move 'east'
+      elsif exits.include?('west') && !exits.include?('southwest') && !exits.include?('east') && !exits.include?('south') && !exits.include?('north') && !exits.include?('southeast') && !exits.include?('northwest') && !exits.include?('northeast')
+        custom_move 'west'
+        custom_move 'west'
+      elsif ((exits.include? 'west') && (exits.include? 'southwest'))
+        custom_move 'west'
+        custom_move 'west'
+        custom_move 'west'
+        custom_move 'west'
+        @done = true
+      end
+    end
+  end
+
+  def finish_task
+    reset_flags
+    @task_count = @task_count + 1
+    get_task
+  end
+
+  def reset_flags
+    Flags.reset('scarecrow')
+    Flags.reset('mice')
+    Flags.reset('weeds')
+    Flags.reset('traps')
+    Flags.reset('grasshoppers')
+    Flags.reset('scream')
+    Flags.reset('landmarks')
+    Flags.reset('tokens')
+    Flags.reset('corn')
+    Flags.reset('halflings')
+    Flags.reset('loot')
+  end
+
+  def custom_move(dir, fast=false)
+    if fast && $frontend != 'genie'
+      put dir
+      @current_type_ahead += 1
+      if @type_ahead_size < @current_type_ahead
+        pause 0.35
+        @current_type_ahead = 0
+      end
+    else
+      move dir
+    end
+  end
+
+  def stow_thing(thing)
+    @settings.cornmaze_containers.each do |container|
+      return if DRC.bput("put #{thing} in my #{container}", 'You put', 'What were you', 'There isn\'t any more room', 'too long') == 'You put'
+    end
+    return if DRC.bput("stow #{thing}", 'You put', 'What were you', 'There isn\'t any more room', 'Stow what') == 'You put'
+    pause 1
+    DRC.message("YOU'VE RUN OUT OF ROOM!  GET SOME MORE SPACE, YOU LAZY SLOB!")
+    DRC.bput("drop #{thing}", 'You drop', 'What were you')
+  end
+end
+
+before_dying do
+  Flags.delete('scarecrow')
+  Flags.delete('mice')
+  Flags.delete('weeds')
+  Flags.delete('traps')
+  Flags.delete('grasshoppers')
+  Flags.delete('scream')
+  Flags.delete('landmarks')
+  Flags.delete('tokens')
+  Flags.delete('corn')
+  Flags.delete('halflings')
+  Flags.delete('loot')
+end
```

Comment:
maybe add maze_done and Task_done, poisoned, and already_built here also, it's pedantic because they'd only be left over if you killed the script but deleting anything that has a Flags.add is the goal.

Reply:
Sarvatt, in the before_dying section or in def reset_flags? Or both?